build/bazelutil/BUILD.bazel

```diff
@@ -10,6 +10,19 @@ genrule(
     stamp = True,
 )
 
+NOGO_FAILED_TEST_OUTPUT = "echo 'must use exactly one of `--config nogo` or `--config nonogo` explicitly'; exit 1"
+
+genrule(
```

Comment:
Combined with the below comment, this test doesn't have to be a genrule; you can make it any kind of rule. Ideally, a rule that wouldn't be confused for having some actual effect - Skylib's https://github.com/bazelbuild/bazel-skylib/blob/main/docs/build_test_doc.md is an excellent option here. Just put the select in any configurable field of build_test.

Reply:
Hmm. I tried the following:

```
build_test(
    name = "test_nogo_configured",
    targets = select({
        "//build/toolchains:nogo": [],
        "//build/toolchains:nonogo_explicit": [],
    }, no_match_error = "must use exactly one of `--config nogo` or `--config nonogo` explicitly"),
)
```

and it failed with the following error:

```
ERROR: Traceback (most recent call last):
	File "/Users/ricky/go/src/github.com/cockroachdb/cockroach/build/bazelutil/BUILD.bazel", line 16, column 11, in <toplevel>
		build_test(
	File "/private/var/tmp/_bazel_ricky/be70b24e7357091e16c49d70921b7985/external/bazel_skylib/rules/build_test.bzl", line 46, column 11, in build_test
		if len(targets) == 0:
Error in len: select is not iterable
ERROR: Skipping '//build/bazelutil:test_nogo_configured': no such target '//build/bazelutil:test_nogo_configured': target 'test_nogo_configured' not declared in package 'build/bazelutil' defined by /Users/ricky/go/src/github.com/cockroachdb/cockroach/build/bazelutil/BUILD.bazel
```

So presumably `targets` is not configurable for `build_test`. I don't know if `build_test` has any other configurable arguments though.

Reply:
Ah, darn u_u Yeah, looks like `build_test` is a macro that tries to introspect its targets. `analysis_test` is just a plain old rule and should be able to handle that, though!

Reply:
OK, done :)